### PR TITLE
Components

### PR DESCRIPTION
**Pull request by MiguelDP4**
- Create books list component
- Create form component
- Create book component
- Create book reducer
- Create book reducer tests
- Create store 
- Pass store as prop to app

### CODE REVIEW

File: src/components/BooksList.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Book from './Book';
+
+function BooksList(props) {
+  const { books } = props;
```
Comment (suretrust) [inline src/components/BooksList.js]: This `books` props should come from the Redux store using `mapStateToProps`.
* [x] it connects to Redux store and fetches books from the state [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/BooksList/ConnetToRedux.png)
* [x] it uses `mapStateToProps` to fetch books from the `store`

File: src/index.js [hunk 2]
```diff
@@ -1,11 +1,51 @@
...
+
+const store = createStore(books, initialBookState);
 
 ReactDOM.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>,
+  <Provider store={store}>
+    <App books={store.getState()} />
+  </Provider>,
```
Comment (suretrust) [inline src/index.js]: You should use `mapStateToProps` to get things from the store, you don't have to pass it directly. Also, you don't need to remove the `React.StrictMode`, you just need to wrap the `App` component with the `Provider` as shown below.
```suggestion
  <React.StrictMode>
    <Provider store={store}>
      <App />
    </Provider>
  </React.StrictMode>,
```

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

- [x] Your actions are missing, let's add them. Example below:
```javascript
export const createBook = (book) => ({
  type: 'CREATE_BOOK',
  book
})
```

- [x] The main reducer should be placed in `src/actions/index.js` and combine app-specific reducers. See what I mean below:

```javascript
import { combineReducers } from 'redux';
import books from './books';

export default combineReducers({ books })
```

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

File: src/components/BooksList.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Book from './Book';
+
```
Comment (shloch) [inline src/components/BooksList.js]: this component should be inside the `src/containers`
folders

from instructions we have 
- 2 `presentational` components (`App` and `Book`) : they should be in the folder `src/components`
- 02 `container` components (`BooksList` and `BooksForm`) : they should be in the folder `src/containers`

So let's just polish this little detail

Comment (shloch) [reply src/components/BooksList.js]: it should connect to redux store
`import { connect } from "react-redux";`

Comment (shloch) [reply src/components/BooksList.js]: It should use `mapStateToProps` to fetch books from the `store`

File: src/components/BookForm.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
```
Comment (shloch) [inline src/components/BookForm.js]: this component should be placed in `src/containers` folder

**Review by shloch — CHANGES_REQUESTED** [review]
## Hello @MiguelDP4 , @fedgut 

We're progressing really good

Just minor things are left as mentioned below

Please let's bring these updates to life

Happy coding

**Review by bellom — APPROVED** [review]
## Hi Team :clap:

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**